pkg/session/session.go

```diff
@@ -208,15 +220,53 @@ func newClient(ctx context.Context, logger logr.Logger, sessionKey string, url *
 	return c, nil
 }
 
-func clearCache(sessionKey string) {
-	sessionMU.Lock()
-	defer sessionMU.Unlock()
-	delete(sessionCache, sessionKey)
+func clearCache(logger logr.Logger, sessionKey string) {
```

Comment:
Am curious, why are we checking if the session is active and calling Logout only on an active session. The `clearCache` method only gets called when we have deemed the session to be not active.
Plus from the `govmomi` code does not seem like calling `Logout` on an inactive session will cause any side effects. IMO we should:
1. Get the tagManager session and if found, call the Logout on it
2. Call logout on the session
3. Delete the session from the cache.

Reply:
Tagmanager.Logout blocks forever in my testing hence keeping the current implmentation